server/utils/originchecker/originchecker.go

```diff
@@ -0,0 +1,100 @@
+package originchecker
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+)
+
+// origins is global for the origin_checker package to be used across all instances of an OriginChecker
```

Comment:
I think it defeats the purpose of having different cors settings for different routers, doesn't it?

Reply:
The global map moved to the `OriginChecker` struct